### ISSUE TITLE
[Firefox] XHRs sent from an incognito window are sent with the cookies from the normal window

### ISSUE THREAD

**Issue by trumad** [Tampermonkey/tampermonkey#816]
## Expected Behavior

XHRs sent in an incognito window with GM.xmlHttpRequest should be sent with the cookies from the incognito instance, not the ones from the non-incognito instance.

## Actual Behavior

You may need to use proxy software (like burp suite) to see the requests in action
1. Visit a site and note down the cookie
2. Create a script which sends a XHR to the site
3. Run the script & note the cookies used during the XHR (should be the same as step 1)
4. Open the same site in an incognito window and note down the cookie
5. Run the script and note the cookies used during the XHR
6. The cookies in step 3 and step 5 will be the same, but they should be different. The cookie from step 5 should match the cookie from step 4, not step 1/3.

If the site requires CSRF tokens, and the script has been crafted to harvest and send the CSRF token as part of the XHR, the CSRF token and cookie will no longer match and script functionality won't work in incognito windows.

## Specifications

- Firefox 70.0.1 (64-bit)
- TM: 4.9.6095
- OS: Mac OS 10.14.6 (18G1012)

## Script

(Please give an example of the script if applicable.)

```js
// ==UserScript==
// @name         xhrBug
// @author       trumad
// @version      0.1
// @description  bug in XHR?
// @match        http://*/*
// @include      *
// @grant        GM.xmlHttpRequest
// @run-at       document-end
// ==/UserScript==

(function() {
    'use strict';

   GM.xmlHttpRequest({
  method: "GET",
  url: `http://${document.domain}/`,
  onload: function(response) {
    console.log(response);
  }
});
})();
```

**Comment by derjanb** [comment]
Requires FF "split" mode support: https://bugzilla.mozilla.org/show_bug.cgi?id=1380812

**Comment by qsniyg** [comment]
@derjanb To be clear, requesting the cookie through the CookieStore for the current tab doesn't fix this issue?

**Comment by derjanb** [comment]
I'm a little torn because the problem is actually caused by Firefox's lack of split mode support.
This means, all responsibility is passed on the extension developers, although split mode support would be the better/all-encompassing solution.
Nevertheless, I will include this workaround for cookies to work as expected, even though everything else (for example the script storage) makes no distinction between normal and incognito mode.

**Comment by trumad** [comment]
Thank you! And thanks for all your hard work on the extension. The internet would be much more difficult to use without it.

**Comment by derjanb** [comment]
Fixed at 4.10.6104
